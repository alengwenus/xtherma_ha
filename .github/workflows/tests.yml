name: Tests

on:
  push:
    branches:
      - main
  pull_request:
  schedule:
    - cron: "0 0 * * *"

env:
  DEFAULT_PYTHON: "3.13"

jobs:
  pre-commit:
    runs-on: "ubuntu-latest"
    name: Pre-commit and pytest
    steps:
      - name: Check out the repository
        uses: actions/checkout@v5

      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
        uses: actions/setup-python@v6
        with:
          python-version: ${{ env.DEFAULT_PYTHON }}
          cache: pip

      - name: Install Python requirements
        run: |
          pip install -r requirements.txt -r dev-requirements.txt

      - name: Run pre-commit on all files
        run: |
          pre-commit run --all-files --show-diff-on-failure --color=always

      - name: Run pytest
        run: |
          python -m pytest --timeout=10 tests

  hacs:
    runs-on: "ubuntu-latest"
    name: HACS
    if: github.repository == 'Xtherma/xtherma_ha'
    steps:
      - name: Check out the repository
        uses: "actions/checkout@v5"

      - name: HACS validation
        uses: "hacs/action@22.5.0"
        with:
          category: "integration"
          ignore: "topics"

  hassfest:
    runs-on: "ubuntu-latest"
    name: Hassfest
    steps:
      - name: Check out the repository
        uses: "actions/checkout@v5"

      - name: Hassfest validation
        uses: "home-assistant/actions/hassfest@master"
