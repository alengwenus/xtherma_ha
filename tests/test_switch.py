"""Tests for the Xtherma switch platform."""

from unittest.mock import patch

import pytest
from homeassistant.components.switch import (
    SwitchEntity,
)
from homeassistant.const import Platform
from homeassistant.exceptions import HomeAssistantError
from pytest_homeassistant_custom_component.common import snapshot_platform

from custom_components.xtherma_fp.xtherma_client_common import XthermaReadOnlyError
from tests.helpers import (
    get_switch_platform,
    provide_modbus_data,
    provide_rest_data,
)

from .conftest import init_integration, init_modbus_integration

SWITCH_ENTITY_ID_450 = "switch.test_entry_xtherma_config_cooling_curve_2_active"
SWITCH_ENTITY_ID_MODBUS_450 = (
    "switch.test_entry_xtherma_modbus_config_cooling_curve_2_active"
)
SWITCH_ENTITY_ID_MODBUS_350 = (
    "switch.test_entry_xtherma_modbus_config_cooling_curve_1_active"
)


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_setup_switch_rest_api(
    hass, entity_registry, snapshot, mock_rest_api_client
) -> None:
    """Test the setup of switch platform using REST API."""
    with patch("custom_components.xtherma_fp._PLATFORMS", [Platform.SWITCH]):
        entry = await init_integration(hass, mock_rest_api_client)

    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)


@pytest.mark.parametrize("mock_modbus_tcp_client", provide_modbus_data(), indirect=True)
async def test_setup_select_modbus_tcp(
    hass, entity_registry, snapshot, mock_modbus_tcp_client
) -> None:
    """Test the setup of switch platform using MODBUS TCP."""
    with patch("custom_components.xtherma_fp._PLATFORMS", [Platform.SWITCH]):
        entry = await init_modbus_integration(hass, mock_modbus_tcp_client)

    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_binary_switch_icon(hass, mock_rest_api_client):
    await init_integration(hass, mock_rest_api_client)
    platform = get_switch_platform(hass)
    entity = platform.entities.get(SWITCH_ENTITY_ID_450)
    assert entity is not None
    assert entity.icon == "mdi:snowflake"


@pytest.mark.parametrize("mock_rest_api_client", provide_rest_data(), indirect=True)
async def test_set_switch_rest(hass, mock_rest_api_client):
    await init_integration(hass, mock_rest_api_client)
    platform = get_switch_platform(hass)
    state = hass.states.get(SWITCH_ENTITY_ID_450)
    assert state.state == "on"
    entity = platform.entities.get(state.entity_id)
    assert entity is not None
    assert isinstance(entity, SwitchEntity)
    # init_integration uses REST-API, which cannot write
    with pytest.raises(HomeAssistantError) as exc_info:
        await entity.async_turn_off()
    assert isinstance(exc_info.value.__cause__, XthermaReadOnlyError)


@pytest.mark.parametrize(
    "mock_modbus_tcp_client",  # This refers to the fixture
    provide_modbus_data(),
    indirect=True,  # This tells pytest to pass the parameter to the fixture
)
async def test_set_switch_modbus(hass, mock_modbus_tcp_client):
    await init_modbus_integration(hass, mock_modbus_tcp_client)
    platform = get_switch_platform(hass)
    state = hass.states.get(SWITCH_ENTITY_ID_MODBUS_450)
    entity = platform.entities.get(state.entity_id)
    assert entity is not None
    assert isinstance(entity, SwitchEntity)
    await entity.async_turn_off()
    kwargs = mock_modbus_tcp_client.write_register.call_args.kwargs
    # verify arguments passed to write_register()
    assert kwargs["address"] == 40
    assert kwargs["value"] == 0
    assert kwargs["slave"] == 1
    await entity.async_turn_on()


@pytest.mark.parametrize(
    "mock_modbus_tcp_client",  # This refers to the fixture
    provide_modbus_data(),
    indirect=True,  # This tells pytest to pass the parameter to the fixture
)
async def test_set_multiple_switches_modbus(hass, mock_modbus_tcp_client):
    await init_modbus_integration(hass, mock_modbus_tcp_client)
    platform = get_switch_platform(hass)
    state = hass.states.get(SWITCH_ENTITY_ID_MODBUS_450)
    entity = platform.entities.get(state.entity_id)
    assert entity is not None
    assert isinstance(entity, SwitchEntity)
    await entity.async_turn_off()
    kwargs = mock_modbus_tcp_client.write_register.call_args.kwargs
    # verify arguments passed to write_register()
    assert kwargs["address"] == 40
    assert kwargs["value"] == 0
    assert kwargs["slave"] == 1
    await entity.async_turn_on()
    state = hass.states.get(SWITCH_ENTITY_ID_MODBUS_350)
    entity = platform.entities.get(state.entity_id)
    assert entity is not None
    assert isinstance(entity, SwitchEntity)
    await entity.async_turn_off()
    kwargs = mock_modbus_tcp_client.write_register.call_args.kwargs
    # verify arguments passed to write_register()
    assert kwargs["address"] == 20
    assert kwargs["value"] == 0
    assert kwargs["slave"] == 1
    await entity.async_turn_on()
