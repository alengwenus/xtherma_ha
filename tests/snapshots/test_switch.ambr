# serializer version: 1
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_cooling_curve_1_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_cooling_curve_1_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:snowflake',
    'original_name': 'Cooling curve 1 active',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '350',
    'unique_id': 'test_entry_xtherma-350',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_cooling_curve_1_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Cooling curve 1 active',
      'icon': 'mdi:snowflake',
      'parameter': '350',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_cooling_curve_1_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_cooling_curve_2_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_cooling_curve_2_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:snowflake',
    'original_name': 'Cooling curve 2 active',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '450',
    'unique_id': 'test_entry_xtherma-450',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_cooling_curve_2_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Cooling curve 2 active',
      'icon': 'mdi:snowflake',
      'parameter': '450',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_cooling_curve_2_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heat_pump_turned_on-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heat_pump_turned_on',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Heat pump turned on',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '001',
    'unique_id': 'test_entry_xtherma-001',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heat_pump_turned_on-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Heat pump turned on',
      'parameter': '001',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heat_pump_turned_on',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heating_curve_1_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heating_curve_1_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:heating-coil',
    'original_name': 'Heating curve 1 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '310',
    'unique_id': 'test_entry_xtherma-310',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heating_curve_1_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Heating curve 1 enabled',
      'icon': 'mdi:heating-coil',
      'parameter': '310',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heating_curve_1_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heating_curve_2_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heating_curve_2_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:heating-coil',
    'original_name': 'Heating curve 2 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '410',
    'unique_id': 'test_entry_xtherma-410',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_heating_curve_2_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Heating curve 2 enabled',
      'icon': 'mdi:heating-coil',
      'parameter': '410',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_heating_curve_2_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_hot_water_now-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_hot_water_now',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water-boiler',
    'original_name': 'Hot water now',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '003',
    'unique_id': 'test_entry_xtherma-003',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_hot_water_now-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config Hot water now',
      'icon': 'mdi:water-boiler',
      'parameter': '003',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_hot_water_now',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_ss14a_enwg_request-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_ss14a_enwg_request',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': '§14a EnWG request',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '815',
    'unique_id': 'test_entry_xtherma-815',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_select_modbus_tcp[mock_modbus_tcp_client0][switch.test_entry_xtherma_modbus_config_ss14a_enwg_request-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_modbus_config §14a EnWG request',
      'parameter': '815',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_modbus_config_ss14a_enwg_request',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_cooling_curve_1_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_cooling_curve_1_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:snowflake',
    'original_name': 'Cooling curve 1 active',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '350',
    'unique_id': 'test_entry_xtherma-350',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_cooling_curve_1_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Cooling curve 1 active',
      'icon': 'mdi:snowflake',
      'parameter': '350',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_cooling_curve_1_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_cooling_curve_2_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_cooling_curve_2_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:snowflake',
    'original_name': 'Cooling curve 2 active',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '450',
    'unique_id': 'test_entry_xtherma-450',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_cooling_curve_2_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Cooling curve 2 active',
      'icon': 'mdi:snowflake',
      'parameter': '450',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_cooling_curve_2_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heat_pump_turned_on-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_heat_pump_turned_on',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Heat pump turned on',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '001',
    'unique_id': 'test_entry_xtherma-001',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heat_pump_turned_on-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Heat pump turned on',
      'parameter': '001',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_heat_pump_turned_on',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heating_curve_1_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_heating_curve_1_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:heating-coil',
    'original_name': 'Heating curve 1 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '310',
    'unique_id': 'test_entry_xtherma-310',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heating_curve_1_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Heating curve 1 enabled',
      'icon': 'mdi:heating-coil',
      'parameter': '310',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_heating_curve_1_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heating_curve_2_enabled-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_heating_curve_2_enabled',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:heating-coil',
    'original_name': 'Heating curve 2 enabled',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '410',
    'unique_id': 'test_entry_xtherma-410',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_heating_curve_2_enabled-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Heating curve 2 enabled',
      'icon': 'mdi:heating-coil',
      'parameter': '410',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_heating_curve_2_enabled',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_hot_water_now-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_hot_water_now',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water-boiler',
    'original_name': 'Hot water now',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '003',
    'unique_id': 'test_entry_xtherma-003',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_hot_water_now-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config Hot water now',
      'icon': 'mdi:water-boiler',
      'parameter': '003',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_hot_water_now',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_ss14a_enwg_request-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.test_entry_xtherma_config_ss14a_enwg_request',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': '§14a EnWG request',
    'platform': 'xtherma_fp',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': '815',
    'unique_id': 'test_entry_xtherma-815',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup_switch_rest_api[mock_rest_api_client0][switch.test_entry_xtherma_config_ss14a_enwg_request-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'test_entry_xtherma_config §14a EnWG request',
      'parameter': '815',
    }),
    'context': <ANY>,
    'entity_id': 'switch.test_entry_xtherma_config_ss14a_enwg_request',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
